koku/api/report/test/tests_queries.py

```diff
@@ -105,3 +114,137 @@ def test_group_data_by_list_missing_units(self):
                     ],
                 }
                 self.assertEqual(expected, out_data)
+
+
+def create_test_handler(mapper=None, params=None):
+    """Create a TestableReportQueryHandler using the supplied args."""
+    if not mapper:
+        mapper = {"filter": {}, "filters": {}}
+
+    if not params:
+        params = {"filter": {}, "group_by": {}, "order_by": {}}
+
+    class TestableReportQueryHandler(ReportQueryHandler):
+        """ A testable minimal implementation of ReportQueryHandler.
+
+             ReportQueryHandler can't be instantiated directly without first setting
+             a few attributes that are required by QueryHandler.__init__().
+         """
+
+        _mapper = Mock(
+            spec=ProviderMap,
+            _report_type_map=Mock(return_value=mapper, get=lambda x, y=None: mapper.get(x, y)),
+            _provider_map=Mock(return_value=mapper, get=lambda x, y=None: mapper.get(x, y)),
+        )
+
+    mock_params = Mock(
+        spec=QueryParameters,
+        parameters=Mock(return_value=params, get=lambda x, y=None: params.get(x, y)),
+        get_filter=lambda x, default: params.get("filter").get(x, default),
+        get_group_by=lambda x, default: params.get("group_by").get(x, default),
+        get=lambda x, default: params.get(x, default),
+        tag_keys=[],
+    )
+    return TestableReportQueryHandler(mock_params)
+
+
+class ReportQueryHandlerTest(TestCase):
+    """Test the report query handler functions."""
+
+    def test_init(self):
+        """Test that we can instantiate a minimal ReportQueryHandler."""
+        rqh = create_test_handler()
+        self.assertIsInstance(rqh, ReportQueryHandler)
+
+    def test_set_operator_specified_filters_and(self):
+        """Test that AND/OR terms are correctly applied to param filters."""
+        operator = "and"
+
+        fake_group = FAKE.word()
+        mapper = {"filter": {}, "filters": {}}
+        params = {"filter": {}, "group_by": {fake_group: ["and:" + FAKE.word(), "and:" + FAKE.word()]}, "order_by": {}}
+
+        rqh = create_test_handler(mapper, params)
+        output = rqh._set_operator_specified_filters(operator)
+        self.assertIsNotNone(output)
+
+        expected = Q(**{})
```

Comment:
Why is the expected here a Q with an empty dict?

Reply:
Mostly because that's what was being returned, regardless of my inputs. 

I'm not sure if that's expected behavior, a bug in the method, or a problem with the inputs I'm using. 

One of the problems with not having unit tests on these functions written by the author of this code is that I'm forced to make some guesses about what should be tested based on the method's return output that I'm observing.